### PR TITLE
perf: optimize Readable.dump

### PR DESCRIPTION
**Pull request by ronag**
## Status

- [ ] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [ ] Tested
- [ ] Benchmarked (**optional**)
- [ ] Documented
- [ ] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
CI is failing

File: lib/api/readable.js [hunk 1]
```diff
@@ -152,34 +154,43 @@ module.exports = class BodyReadable extends Readable {
...
+        ? util.addAbortListener(signal, () => {
+          this.destroy()
+        })
+        : noop
+
+      this
+        .on('close', function () {
+          signalListenerCleanup()
+          if (signal?.aborted) {
+            reject(signal.reason || Object.assign(new Error('The operation was aborted'), { name: 'AbortError' }))
```
Comment (Ethan-Arrowood) [inline lib/api/readable.js:180]: Is the `Object.assign` piece faster or better than us creating a `class AbortError extends Error` that we then just instantiate?

**Review by Ethan-Arrowood — APPROVED** [review]

File: lib/api/readable.js (hunk 1 above)
Comment (ronag) [reply lib/api/readable.js:180]: not really, just convenience...

File: lib/api/readable.js [hunk 2]
```diff
@@ -152,34 +154,43 @@ module.exports = class BodyReadable extends Readable {
...
-    } catch {
-      util.throwIfAborted(signal)
-    } finally {
-      if (typeof signalListenerCleanup === 'function') {
-        signalListenerCleanup()
-      } else if (signalListenerCleanup) {
-        signalListenerCleanup[Symbol.dispose]()
-      }
+
+    if (this.closed) {
```
Comment (metcoder95) [inline lib/api/readable.js:165]: Should we save the extra `Promise` made by the `async` keyword?
The returned promise seems to handle already the required `dump` logic

Comment (ronag) [reply lib/api/readable.js:165]: Good catch!

**Review by metcoder95 — APPROVED** [review]
LGTM 👍

**Pull request merged**